src/main/java/org/apache/sling/jcr/base/internal/LoginAdminAllowlistConfigurationEventListener.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sling.jcr.base.internal;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.osgi.service.cm.ConfigurationEvent;
+import org.osgi.service.cm.ConfigurationListener;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component
+public class LoginAdminAllowlistConfigurationEventListener implements ConfigurationListener {
```

Comment:
Could we register a single `ConfigurationListener`?

Maybe `ConfigurationUpdater` can be turned into a POJO that is parameterized with the PIDs and properties to replace. Then it could be called with `ConfigurationEvent` and `ConfigurationAdmin` as its arguments.

Your `ConfigurationListener` could then pass all events to both `ConfigurationUpdater` instances, which would only handle events for the PID they are parameterized for.